library/network_connections.py

```diff
@@ -851,11 +851,51 @@ def get_default_vlan(self):
             'id': None,
         }
 
+class ArgValidator_DictMacvlan(ArgValidatorDict):
+
+    VALID_MODES = ['vepa', 'bridge', 'private', 'passthru', 'source']
+
+    def __init__(self):
+        ArgValidatorDict.__init__(self,
+            name = 'macvlan',
+            nested = [
+                ArgValidatorStr ('mode', enum_values = ArgValidator_DictMacvlan.VALID_MODES, default_value = 'bridge'),
+                ArgValidatorNum ('mode_id', val_min = 1, val_max = 5, default_value = None),
```

Comment:
I am totally with you. I only introduced it because networkmanager wants the number, not the string. Which then begets the question whether the mapping string -> integer is fixed across systems. That's why I used Util.NM() below. 
I see two ways to change this (after I remove mode_id):
- use fixed mapping mode string to mode id. But are the numbers fixed...?
- Use the int(getattr( NM.SettingMacvlanMode, mode.upper() )), but place it in create_connection(), because there NM is already available.
I'd prefer the second choice, but code like that is no longer present in create_connection, that's why I wasn't sure if it is acceptable.